File: scripts/algorithms/M/Maximum Ascending Subarray Sum/Maximum Ascending Subarray Sum.cpp
// Runtime: 0 ms (Top 100.00%) | Memory: 8.5 MB (Top 42.32%)
class Solution {
public:
    int maxAscendingSum(vector<int>& nums) {
        int max_sum = nums[0], curr = nums[0];
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i-1] < nums[i]) {
                curr += nums[i];
            }
            else {
                max_sum = max(max_sum, curr);
                curr = nums[i];
            }
        }
        return max(max_sum, curr);
    }
};

File: scripts/algorithms/M/Maximum Ascending Subarray Sum/Maximum Ascending Subarray Sum.js
// Runtime: 99 ms (Top 29.63%) | Memory: 41.6 MB (Top 91.36%)
var maxAscendingSum = function(nums) {
    const subarray = nums.reduce((acc, curr, index) => {
        curr > nums[index - 1] ? acc[acc.length - 1] += curr : acc.push(curr);
        return acc;
    }, []);

    return Math.max(...subarray);
};